src/wp/SplashScreen.cs

```diff
@@ -68,27 +77,52 @@ public override void OnInit()
             if (!WasShown)
             {
                 WasShown = true;
-                show();
+                this.show();
             }
         }
 
-        void LoadConfigValues()
+        private static void GetPreference(XDocument document, string preferenceName, Action<string> action)
```

Comment:
This method works somewhat backwards, where you check up to N strings for every preference, making your worst-case performance O(n^2) (in the case that you don't find the preferences).  Suggested alternative: Populate a Dictionary<string, Action<string>>, and then for-each preference element, if (dictionary.TryGetValue(preferenceName, out action)) action(value).  Dictionary lookups approach O(1).